approval_polls/views.py

```diff
@@ -79,7 +79,7 @@ def add(request):
     if not len(choices):
       return render(request, 'approval_polls/error.html', {'error_message': 'At least one choice is required'})
 
-    p = Poll(question=question, pub_date=timezone.now())
+    p = Poll(question=question, pub_date=timezone.now(), username=request.POST['username'].strip())
```

Comment:
@Brianjr0428 My understanding is that we need the user object associated with the poll in two cases

1. When creating the poll. In this case, the user is available in the form of `request.user` as you've found out from your research. There's no need to query the database in this case.
2. When displaying the username associated with the user who created the poll. In this case the username can be accessed using something like `poll.user.username`

Is there any case I'm missing?